scripts/spack_recipes/build_lbann.sh

```diff
@@ -1,4 +1,7 @@
-#!/bin/sh
+#!/bin/bash
+
+. $bamboo_SPACK_ROOT/share/spack/setup-env.sh
```

Comment:
This will work for Bamboo but not general use. If `bamboo_SPACK_ROOT` exists, then run this line. 

That will mean we don't have to pass in an argument like `inBamboo`.

Reply:
`if [ ! -z "$bamboo_SPACK_ROOT" ];` then run this line